view/sharedcache/core/SharedCache.cpp

```diff
@@ -2779,6 +2779,31 @@ std::vector<Ref<Symbol>> SharedCache::ParseExportTrie(std::shared_ptr<MMappedFil
 	return symbols;
 }
 
+
+std::vector<std::pair<uint64_t, std::pair<BNSymbolType, std::string>>> SharedCache::GetExportListForHeader(SharedCacheMachOHeader header, std::function<std::shared_ptr<MMappedFileAccessor>()> provideLinkeditFile)
```

Comment:
I think this should be possible as once a vector is created it shouldn't be reallocated by `m_exportInfos`?

Reply:
The thread safety problems arise when `m_exportInfos` is modified. If the hash table needs to grow it will reallocate and moves its contents. This would invalidate any existing references to data it stores.

One solution that doesn't require any additional locking would be to add a layer of indirection: have `m_exportInfos` store a `std::unique_ptr` to the vector. Then the vector itself will never move, only the pointer that owns it. You could then safely return a reference to the pointed-at vector from this function.

Reply:
I'm not sure using `std::unique_ptr` is pratical here because `m_exportInfos` in `ViewStateCacheStore` causes compiler errors due to copying mechanics when `m_exportInfos` has a `unique_ptr` type. `shared_ptr` works fine and I don't see much of an issue in using that instead